### PR TITLE
feat(synthetics): syn-nodejs-puppeteer-9.0 as supported runtime

### LINKED ISSUE #31271: aws-synthetics: support syn-nodejs-puppeteer-9.0 runtime
### Describe the feature

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Library_nodejs_puppeteer.html confirms that Synthetics runtime "syn-nodejs-puppeteer-9.0" is available.

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts only has up to 8.0

Can we support this in CDK please? Thanks.

### Use Case

Need to be able to use latest runtimes

### Proposed Solution

Update aws-cdk-lib/aws-synthetics/lib/runtime.ts

### Other Information

Previous PR for runtime updating: #30851 

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.155.0

### Environment details (OS name and version, etc.)

n/a

### PR DESCRIPTION
**Pull request by ashishdhingra**
### Issue # (if applicable)

Closes #31271.

### Reason for this change
[Runtime versions using Node.js and Puppeteer](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Library_nodejs_puppeteer.html) confirms that Synthetics runtime `syn-nodejs-puppeteer-9.0` is available.

### Description of changes
Added `syn-nodejs-puppeteer-9.0` as supported runtime.

### Description of how you validated changes
N/A

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts [hunk 1]
```diff
@@ -221,6 +221,19 @@ export class Runtime {
...
   public static readonly SYNTHETICS_NODEJS_PUPPETEER_8_0 = new Runtime('syn-nodejs-puppeteer-8.0', RuntimeFamily.NODEJS);
 
+  /**
+   * `syn-nodejs-puppeteer-9.0` includes the following:
+   * - Lambda runtime Node.js 20.x
+   * - Puppeteer-core version 22.12.1
+   * - Chromium version 126.0.6478.126
+   *
+   * New Features:
+   * - **Bug fixes** Bug fix to enable visual monitoring capabilities.
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts:231]: Nit:

```
* Bug fixes:
* - **Bug fix**: Enable visual monitoring capabilities.
```

Comment (ashishdhingra) [reply packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts:231]: @msambol Thanks for the review. I copied the content from https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Library_nodejs_puppeteer.html#CloudWatch_Synthetics_runtimeversion-nodejs-puppeteer-9.0.

Comment (msambol) [reply packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts:231]: right, I'm just saying it shouldn't be under `New Features:`

Comment (kaizencc) [reply packages/aws-cdk-lib/aws-synthetics/lib/runtime.ts:231]: I think this is fine -- its a little weird but we've already established a pattern of `bug fixes` being listed as a bullet under `new features`.

**Review by kaizencc — APPROVED** [review]

**Review by kaizencc — APPROVED** [review]

**Pull request merged**